src/tests/multihost/ad/test_access_control.py

```diff
@@ -0,0 +1,414 @@
+
+    @staticmethod
+    def test_simple_deny_user_to_invalid_user(multihost, create_aduser_group, backupsssdconf):
+        """
+        :title: Set simple_deny_user to an invalid user
+        :id: 76aea575-420b-4276-b0c6-383a4a8499a0
+        :description: Setting simple_deny_user to an invalid user would allow all users to log in
+        :steps:
+            1. Set 'simple_deny_users' to an invalid AD-user
+            2. Restart SSSD
+            3. Log in with any AD-user
+        :expectedresults:
+            1. Option should be correctly set
+            2. SSSD should start correctly
+            3. Any ADuser should be able to log in
+        """
+        tools = sssdTools(multihost.client[0])
+        (aduser, _) = create_aduser_group
+        domain_name = tools.get_domain_section_name()
+        dom_section = f'domain/{domain_name}'
+        sssd_params = {'access_provider': 'simple',
+                       'simple_deny_users': f'nonuser@{domain_name}'}
+        tools.sssd_conf(dom_section, sssd_params, action='add')
+        tools.clear_sssd_cache()
+        ret = ssh_login(multihost, f'{aduser}@{domain_name}')
+        assert ret == 'Success', 'ADuser log in failed'
+
+    @staticmethod
+    def test_simple_allow_groups_top_nested(multihost, create_aduser_group, create_nested_group, backupsssdconf):
+        """
+        :title: Set simple allow groups to the top-level nested group
+        :id: ae19a604-1ce4-4a03-b1a6-4023aa71961f
+        :description: Set simple_allow_groups to a top-level nested group. This top-level group will have one group
+         as it's member. The member-group has one AD-user as group-member. The AD-user from group should
+         be able to log in.
+        :Steps:
+            1. Create two groups.
+            2. Add one group as a member of other group.
+            3. Create an AD user and add that user as member of a member group
+            4. Set 'simple_allow_groups' to a top level nested group
+            5. Restart SSSD
+            6. Log in with Allowed AD-user
+        :expectedresults:
+            1. Option should be correctly set
+            2. SSSD should start correctly
+            3. Should succeed
+            4. Should succeed
+            5. Should succeed
+            6. Should succeed
+        """
+        run_id = create_nested_group
+        aduser = f'testuser-{run_id}'
+        l1_grp = f'testgrp-l1-{run_id}'
+        tools = sssdTools(multihost.client[0], multihost.ad[0])
+        domain_name = tools.get_domain_section_name()
+        dom_section = f'domain/{domain_name}'
+        sssd_params = {'access_provider': 'simple',
+                       'simple_allow_groups': f'{l1_grp}@{domain_name}'}
+        tools.sssd_conf(dom_section, sssd_params, action='add')
+        tools.clear_sssd_cache()
+        ret = ssh_login(multihost, f'{aduser}@{domain_name}')
+        assert aduser in ret, 'ADuser log in failed'
```

Comment:
Hi,

this test fails for me since `ssh_login()` does not return the `id` output anymore, so the assert should be adopted to the new output of `ssh_login()`.

bye,
Sumit

Reply:
rectified the assertion.